src/EnergyPlus/WeatherManager.cc

```diff
@@ -3546,7 +3547,7 @@ namespace WeatherManager {
                                             RField21);
 
             if (!succeeded) ErrorInterpretWeatherDataLine(state, WYear, WMonth, WDay, WHour, WMinute, Line, current_line);
-            current_line.remove_prefix(nth_pos + 1);
+            current_line.remove_prefix(nth_pos);
```

Comment:
The actual fix is here. Before it would get to `0,909999999` end up choping it to `09999999` instead of `909999999`so it'd think only **8** weather codes were given.

Reply:
Got it